thorlcr/graph/thgraphmaster.cpp

```diff
@@ -656,6 +650,58 @@ void CMasterActivity::updateFileWriteCostStats(IFileDescriptor & fileDesc, IProp
     }
 }
 
+unsigned CMasterActivity::setupSpace4FileStats(const char *normalizedFilename, bool reInit, std::vector<int> & fileStatsTable, std::vector<OwnedPtr<CThorStatsCollection>> & fileStats, bool isSuper, unsigned numSubs)
+{
```

Comment:
>Related/pre-existing issue - lookupReadFile should not be re-adding a file to readFiles if it is already there (it can look up in map 1st).
In fact, I think readFiles and fileStats will go out of sync., if some of the time the same files are read at the moment (and cause problems in updateFileReadCostStats.

My mistake, that's an existing issue. lookupReadFile checks the map first (via findReadFile), so will not re-lookup/re-add.